core/templates/dev/head/components/top_navigation_bar/TopNavigationBarDirective.js

```diff
@@ -0,0 +1,105 @@
+ */
+
+oppia.directive('topNavigationBar', [function() {
+  return {
+    restrict: 'E',
+    scope: {},
+    templateUrl: 'components/topNavigationBar',
+    controller: [
+      '$scope', '$http', '$window', '$timeout', 'UrlInterpolationService',
+      'SidebarStatusService', 'LABEL_FOR_CLEARING_FOCUS',
+      'siteAnalyticsService', 'windowDimensionsService',
+      function(
+          $scope, $http, $window, $timeout, UrlInterpolationService,
+          SidebarStatusService, LABEL_FOR_CLEARING_FOCUS,
+          siteAnalyticsService, windowDimensionsService) {
+        $scope.NAV_MODE = GLOBALS.NAV_MODE;
+        $scope.LABEL_FOR_CLEARING_FOCUS = LABEL_FOR_CLEARING_FOCUS;
+        $scope.getStaticImageUrl = UrlInterpolationService.getStaticImageUrl;
+
+        $scope.onLoginButtonClicked = function(loginUrl) {
+          siteAnalyticsService.registerStartLoginEvent('loginButton');
+          $timeout(function() {
+            $window.location = loginUrl;
+          }, 150);
+          return false;
+        };
+
+        $scope.profileDropdownIsActive = false;
+        $scope.onMouseoverProfilePictureOrDropdown = function(evt) {
+          angular.element(evt.currentTarget).parent().addClass('open');
+          $scope.profileDropdownIsActive = true;
+        };
+        $scope.onMouseoutProfilePictureOrDropdown = function(evt) {
+          angular.element(evt.currentTarget).parent().removeClass('open');
+          $scope.profileDropdownIsActive = false;
+        };
+
+        $scope.onMouseoverDropdownMenu = function(evt) {
+          angular.element(evt.currentTarget).parent().addClass('open');
+        };
+        $scope.onMouseoutDropdownMenu = function(evt) {
+          angular.element(evt.currentTarget).parent().removeClass('open');
+        };
+
+        if (GLOBALS.userIsLoggedIn) {
+          // Show the number of unseen notifications in the navbar and page
+          // title, unless the user is already on the dashboard page.
+          $http.get('/notificationshandler').then(function(response) {
+            var data = response.data;
+            if ($window.location.pathname !== '/') {
+              $scope.numUnseenNotifications = data.num_unseen_notifications;
+              if ($scope.numUnseenNotifications > 0) {
+                $window.document.title = (
+                  '(' + $scope.numUnseenNotifications + ') ' +
+                  $window.document.title);
+              }
+            }
+          });
+        }
+
+        $scope.windowIsNarrow = windowDimensionsService.isWindowNarrow();
+        windowDimensionsService.registerOnResizeHook(function() {
+          $scope.windowIsNarrow = windowDimensionsService.isWindowNarrow();
+          $scope.$apply();
+          // Close the sidebar, if necessary.
+          SidebarStatusService.closeSidebar();
+        });
+
+        $scope.toggleSidebar = SidebarStatusService.toggleSidebar;
+
+        // TODO(sll): Sort out (Jinja):
```

Comment:
ng-if sounds sensible, yes.

And yes, using GLOBALS seems like a good idea. Can't think of a good reason against it, and using it does remove the need for an extra request.